material-ui/material-ui.d.ts

```diff
@@ -7228,7 +7228,7 @@ declare module 'material-ui/utils/withWidth' {
       mediumWidth?: number;
       resizeInterval?: number;
   }
-  export default function withWidth(options?: Options): __React.ComponentClass<any>
+  export default function withWidth<C extends __React.ComponentClass<any> | __React.StatelessComponent<any>>(options?: Options): (component: C) => C;
```

Comment:
I guess this will be sufficient:
```
export default function withWidth(options?: Options): <C extends Function>(component: C) => C;
```

Here are some examples of HOC currently in dt:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/2554570b9211844703361c90755c81fefe12eb86/redux-form/redux-form.d.ts#L295
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/7e656a65e70709d8d8068b6dff1875155e44166e/react-intl/react-intl.d.ts#L17